week09/movie/src/pages/HomePage.jsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import styled from "styled-components";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import Movie from "../components/Movie";
+import Loader from "../components/Loader";
+
+const Container = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+`;
+
+const SortContainer = styled.div`
+    width: 100vw;
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+`;
+
+const SortList = styled.div`
+    margin-top: 10px;
+    margin-right: 30px;
+`;
+
+const MovieList = styled.div`
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+`;
+
+const StyledLink = styled(Link)`
+    text-decoration: none;
+    color: inherit;
+`;
+
+const Text = styled.div`
+    font-size: 40px;
+    margin-left: 15px;
+    margin-top: 5px;
+`;
+
+const HomePage = () => {
+    const [loading, setLoading] = useState(true);
+    const [movies, setMovies] = useState([]);
+    const [sort, setSort] = useState("download_count");
+
+    useEffect(() => {
+        async function getMovies() {
+            try {
+                const res = await axios.get(
+                    `https://yts.mx/api/v2/list_movies.json?limit=50&sort_by=${sort}`
+                );
+                setMovies(res.data.data.movies);
+                setLoading(false);
+                console.log(movies);
+            } catch (error) {
+                console.log(error);
+            }
+        }
+        getMovies();
+    }, [sort]);
+
```

Comment:
- [x] 3. 종속성 배열 movie
해당 useEffect에서 sort 뿐만 아니라 movies 상태를 참조하고 있으나 의존성 배열에 추가하면 무한 루프의 위험에 빠지므로, 역시 여기서 `console.log(movies);`를 제거 후 따로 분리하여 작성합니다.
```suggestion
           } catch (error) {
        console.log(error);
      }
    }
    getMovies();
  }, [sort]);

  useEffect(() => {
    console.log(movies);
  }, [movies]);
```